js/admin/Forms.tsx

```diff
@@ -487,4 +488,93 @@ export class Timeago extends React.Component<{ time: Date }> {
     render() {
         return this.props.time ? timeago.format(this.props.time) : ""
     }
+}
+
+import TagBadge, { Tag } from './TagBadge'
+
+// NOTE (Mispy): Using my own fork of this which is modified to autoselect the first option.
+// Better UX for case when you aren't adding new tags, only selecting from list.
+const ReactTags = require('react-tag-autocomplete')
+
+@observer
+class EditTags extends React.Component<{ tags: Tag[], suggestions: Tag[], onDelete: (index: number) => void, onAdd: (tag: Tag) => void, onSave: () => void }> {
+    dismissable: boolean = true
+
+    @action.bound onClickSomewhere(e: MouseEvent) {
+        if (this.dismissable)
+            this.props.onSave()
+        this.dismissable = true
+    }
+
+    @action.bound onClick() {
+        this.dismissable = false
+    }
+
+    componentDidMount() {
+        document.addEventListener('click', this.onClickSomewhere)
+    }
+
+    componentWillUnmount() {
+        document.removeEventListener('click', this.onClickSomewhere)
+    }
+
+    render() {
+        const {tags, suggestions} = this.props
+        return <div className="EditTags" onClick={this.onClick}>
+            <ReactTags tags={tags} suggestions={suggestions} handleAddition={this.props.onAdd} handleDelete={this.props.onDelete} minQueryLength={1}/>
+        </div>
+    }
+}
+
+@observer
+export class EditableTags extends React.Component<{ tags: Tag[], suggestions: Tag[], onSave: (tags: Tag[]) => void, disabled?: boolean }> {
+    @observable isEditing: boolean = false
+    base: React.RefObject<HTMLDivElement> = React.createRef()
+
+    @observable tags: Tag[] = _.clone(this.props.tags)
+
+    @action.bound onAddTag(tag: Tag) {
+        this.tags.push(tag)
+        this.tags = _.uniq(this.tags).filter(t => t.name !== 'Uncategorized')
```

Comment:
...just testing out this new GitHub feature 😄

```suggestion
        this.tags = _.uniq(this.tags).filter(t => t.name !== 'Uncategorized')
        this.ensureUncategorized()
```